File: VB/CustomBarSubItem/CustomBarManager/CustomBarLinkPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraBars.Painters
Imports DevExpress.XtraBars.Styles

Namespace CustomBar.CustomBarManager
	Friend Class CustomBarLinkPainter
		Inherits BarCustomContainerLinkPainter
		Public Sub New(ByVal paintStyle As BarManagerPaintStyle)
			MyBase.New(paintStyle)

		End Sub

		Protected Overrides Sub DrawLinkHorizontal(ByVal e As BarLinkPaintArgs)
			Dim item As CustomBarSubItem = TryCast(e.LinkInfo.Link.Item, CustomBarSubItem)
			If item.IsChildItemChecked Then
				MyBase.DrawLinkPressed(e)
			Else
				MyBase.DrawLinkHorizontal(e)
			End If
		End Sub
	End Class
End Namespace

File: CS/CustomBarSubItem/CustomBarManager/CustomBarLinkPainter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraBars.Painters;
using DevExpress.XtraBars.Styles;

namespace CustomBar.CustomBarManager {
    class CustomBarLinkPainter : BarCustomContainerLinkPainter{
        public CustomBarLinkPainter(BarManagerPaintStyle paintStyle)
            : base(paintStyle) {
            
        }

        protected override void DrawLinkHorizontal(BarLinkPaintArgs e) {
            CustomBarSubItem item = e.LinkInfo.Link.Item as CustomBarSubItem;
            if (item.IsChildItemChecked )
                base.DrawLinkPressed(e);
            else
                base.DrawLinkHorizontal(e);
        }
    }
}
